### PR TITLE
#11072 Show task variables before running it

### LINKED ISSUE #11072: Request variable values before execution
When a task has defined variables, these variables has to be set in the task definition. It would be great, when launching a tasks with variables, to have a screen prompting to validate or change the values of these variables.

So when defining a variable, I would expect to have more parameter to a variable:
- readonly 
   - true: variable is set in the task definition
   - false: the variable is prompted in the screen before execution
- pattern: regex to validate the input. This can be improved with custom input field (date, number, ...) but that can also be a pain to convert later...
- help message: a brief description of the parameter to help the user to write the variable value

### PR DESCRIPTION
**Pull request by ShadelessFox**
image

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/TaskHandlerRun.java [hunk 1]
```diff
@@ -85,4 +93,25 @@ public void updateElement(UIElement element, Map parameters) {
         }
 
     }
+
+    private static boolean promptTaskVariables(@NotNull DBTTask task) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/TaskHandlerRun.java]: I'd prefer to call it like `confirmTaskVariables` as we stop task execution if user press Cancel or close the dialog.

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/TaskHandlerRun.java [hunk 2]
```diff
@@ -85,4 +93,25 @@ public void updateElement(UIElement element, Map parameters) {
...
+        if (task.getType().supportsVariables()) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/TaskHandlerRun.java]: IMHO it's better to check it before calling this method. Like 
```
if (task.getType().supportsVariables() && !confirmTaskVariables(task)) {
    return;
}
```

File: plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/TaskHandlerRun.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/view/TaskHandlerRun.java]: It's sounds a bit odd to me but okay.

**Review by E1izabeth — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by DariaMarkaryan** [comment]
Verified

**Pull request merged**